playground/add-visa-processing-component

singhvishvendra700 opened the pull request:
Playground: Visa Processing
This PR adds a new Playground for Visa Processing, allowing users to seamlessly explore visa-related information.

Functionality:
- Users can search and select a country (fetched via the `GET Countries` API).
- Upon selecting a country, the app retrieves available visa types using the `GET VISA_TYPES` API.
- When a visa type is selected, the app fetches and displays the visa requirements via the `VISA_TYPES/REQUIREMENTS API` API.

Reference Images:
<img width="1470" alt="image-1" src="https://github.com/user-attachments/assets/cb4e1347-887e-4a6f-9a86-07e197e8aa7f" />
<img width="1470" alt="image-2" src="https://github.com/user-attachments/assets/2b1fca09-0785-4a99-8bf8-1c4a3d6e0073" />
<img width="1470" alt="image-3" src="https://github.com/user-attachments/assets/a53e2c16-7c71-4307-9221-0740b95fc163" />
<img width="1470" alt="image-4" src="https://github.com/user-attachments/assets/c2ea900c-3900-4efe-9e2e-d206c72fa7a9" />
<img width="1470" alt="image-5" src="https://github.com/user-attachments/assets/0fd9d146-498c-4734-b157-cd46dc984fd8" />

girishsawant999 commented on src/pages/PlayGround/playgrounds/VisaProcessing/hooks.ts:
```diff
@@ -0,0 +1,110 @@
+  getVisaRequirements,
+} from "../../../../utils/api";
+
+const useVisaProcessing = () => {
```
Consider splitting this hook into multiple reusable hooks to enhance modularity and maintainability. This will make the code more readable, easier to test, and improve reusability across components.

singhvishvendra700 replied:
Thank you! I've updated the code accordingly.

girishsawant999 commented on src/utils/api.ts:
```diff
@@ -34,3 +34,57 @@ export const searchOrders = async (
+  const { server, token } = authData;
+  const { pageNo, pageSize } = data;
+  const request: AxiosRequestConfig = {
+    url: `${server}/api/v1/countries?page_no=${pageNo}&page_size=${pageSize}`,
```
Consider creating an Axios instance with a predefined base URL and using it throughout the project instead of directly using Axios. This will improve maintainability, enable easier configuration (e.g., adding headers, interceptors), and ensure consistency across API calls.

girishsawant999 commented on src/utils/api.ts:
```diff
@@ -34,3 +34,57 @@ export const searchOrders = async (
+export const getCountryDetailsByCountry = async (
+  authData: TAuthData,
+  data: { symbol: string }
+): Promise<unknown> => {
```
Use an Axios instance with a base URL and add type safety for API responses (e.g., ```axios.get<Type>```).

girishsawant999 commented:
@singhvishvendra700  Improve the UI for better clarity and usability.

singhvishvendra700 commented:
@girishsawant999  - I have refactored hooks into smaller reusable functions to enhance modularity and maintainability. Implemented a centralized Axios instance for consistent API calls and easier configuration. Also, Added TypeScript type safety for API responses to prevent runtime errors. 

I have tried changing UI to for better clarity and usability. Attaching the images below:
<img width="1470" alt="Screenshot 2025-03-14 at 11 31 12 PM" src="https://github.com/user-attachments/assets/9592b3ae-d107-48b6-9d6e-d94635eaf006" />
